spec/integration/github_omniauth_spec.rb

```diff
@@ -0,0 +1,232 @@
+        }
+      )
+
+    stub_request(:get, "https://api.github.com/user")
+      .with(
+        headers: {
+          "Authorization" => "Bearer #{access_token}"
+        }
+      )
+      .to_return(
+        status: 200,
+        body: JSON.dump(
+          login: "octocat",
+          id: 1,
+          node_id: "MDQ6VXNlcjE=",
+          avatar_url: "https://github.com/images/error/octocat_happy.gif",
+          gravatar_id: "",
+          url: "https://api.github.com/users/octocat",
+          html_url: "https://github.com/octocat",
+          followers_url: "https://api.github.com/users/octocat/followers",
+          following_url: "https://api.github.com/users/octocat/following{/other_user}",
+          gists_url: "https://api.github.com/users/octocat/gists{/gist_id}",
+          starred_url: "https://api.github.com/users/octocat/starred{/owner}{/repo}",
+          subscriptions_url: "https://api.github.com/users/octocat/subscriptions",
+          organizations_url: "https://api.github.com/users/octocat/orgs",
+          repos_url: "https://api.github.com/users/octocat/repos",
+          events_url: "https://api.github.com/users/octocat/events{/privacy}",
+          received_events_url: "https://api.github.com/users/octocat/received_events",
+          type: "User",
+          site_admin: false,
+          name: "monalisa octocat",
+          company: "GitHub",
+          blog: "https://github.com/blog",
+          location: "San Francisco",
+          email: "octocat@github.com",
+          hireable: false,
+          bio: "There once was...",
+          twitter_username: "monatheoctocat",
+          public_repos: 2,
+          public_gists: 1,
+          followers: 20,
+          following: 0,
+          created_at: "2008-01-14T04:33:35Z",
+          updated_at: "2008-01-14T04:33:35Z",
+          private_gists: 81,
+          total_private_repos: 100,
+          owned_private_repos: 100,
+          disk_usage: 10000,
+          collaborators: 8,
+          two_factor_authentication: true,
+          plan: {
+            name: "Medium",
+            space: 400,
+            private_repos: 20,
+            collaborators: 0
+          }
+        ),
+        headers: {
+          "Content-Type" => "application/json"
+        }
+      )
+  end
+
+  it "doesn't sign in anyone if none of the emails from github are verified" do
+    post "/auth/github"
+
+    setup_github_emails_stub(
+      [
+        {
+          email: user1.email,
+          primary: true,
+          verified: false,
+          visibility: "private"
+        },
+        {
+          email: user2.email,
+          primary: false,
+          verified: false,
+          visibility: "private"
+        }
+      ]
+    )
+
+    post "/auth/github/callback", params: {
+      state: session["omniauth.state"],
+      code: temp_code
+    }
+    expect(response.status).to eq(302)
+    expect(response.location).to eq("http://test.localhost/")
+    expect(cookies[:_t]).to be_blank
```

Comment:
Small note you can rely on `session[:current_user_id]` to determine if a user is logged in or not

https://github.com/discourse/discourse/blob/d1c5529aadda4942e4981f4fd9f7f9699dc9e231/spec/rails_helper.rb#L405

Reply:
Awesome, thanks! Updated the specs to use `session[:current_user_id]`.